public/main.js

```diff
@@ -117,53 +151,18 @@ function findTodo(id) {
     }
 }
 
-function updateTodo(element, todoId, callback) {
-    var updateRequest = new XMLHttpRequest();
-    var  todo = findTodo(todoId);
-    updateRequest.open("PUT", "/api/todo/" + todo.id);
-    updateRequest.setRequestHeader("Content-type", "application/json");
-    var d = element.checked;
-    updateRequest.send(JSON.stringify({
-        title: todo.title,
-        isComplete: element.checked
-    }));
-    updateRequest.onload = function() {
-        if (this.status === 200) {
-            if (element.checked) {
-                element.parentNode.className += "completed";
-            } else {
-                element.parentNode.className = element.parentNode.className.replace("completed", "");
-            }
-            callback();
-        } else {
-            renderMessageDialog("error", "Failed to update item " +
-                todoId.toString() + ". Server returned " + this.status + " - " + this.responseText);
-        }
-    };
-}
-
-function deleteTodo(todoId, callback) {
-    var deleteRequest = new XMLHttpRequest();
-    deleteRequest.open("DELETE", "/api/todo/" + todoId);
-    deleteRequest.onload = function() {
-        if (this.status === 200) {
-        } else {
-            renderMessageDialog("error", "Failed to delete item " +
-                todoId.toString() + ". Server returned " + this.status + " - " + this.responseText);
-        }
-        callback(this.status);
-    };
-    deleteRequest.send();
+function checkStatus(response) {
+    if (response.status >= 200 && response.status < 300) {
```

Comment:
Took that whole function from the pollyfill fetch examples :worried: [here](https://github.com/github/fetch#handling-http-error-statuses). I agree with you on 300s I'll change that. Tested without the function and fetch doesn't seem to be throwing errors on it's own